pkg/validation/internal/good_practices.go

```diff
@@ -69,9 +78,37 @@ func validateResourceRequests(csv *operatorsv1alpha1.ClusterServiceVersion) (err
 					"to ensure the resource request for CPU and Memory. Be aware that for some clusters configurations "+
 					"it is required to specify requests or limits for those values. Otherwise, the system or quota may "+
 					"reject Pod creation. More info: https://master.sdk.operatorframework.io/docs/best-practices/managing-resources/", c.Name)
-				warns = append(warns, msg)
+				checks.warns = append(checks.warns, msg)
 			}
 		}
 	}
-	return errs, warns
+	return checks
+}
+
+// checkBundleName will validate the operator bundle name informed via CSV.metadata.name.
+// The motivation for the following check is to ensure that operators authors knows that operator bundles names should
+// follow a name and versioning convention
+func checkBundleName(checks CSVChecks) CSVChecks {
+
+	// Check if is following the semver
+	re := regexp.MustCompile("([0-9]+)\\.([0-9]+)\\.([0-9]+)(?:-([0-9A-Za-z-]+(?:\\.[0-9A-Za-z-]+)*))?(?:\\+[0-9A-Za-z-]+)?$")
+	match := re.FindStringSubmatch(checks.csv.Name)
+
+	if len(match) > 0 {
+		if _, err := semver.Parse(match[0]); err != nil {
+			checks.warns = append(checks.warns, fmt.Errorf("csv.metadata.Name %v is not following the versioning "+
+				"convention (MAJOR.MINOR.PATCH e.g 0.0.1): https://semver.org/", checks.csv.Name))
+		}
+	} else {
+		checks.warns = append(checks.warns, fmt.Errorf("csv.metadata.Name %v is not following the versioning "+
+			"convention (MAJOR.MINOR.PATCH e.g 0.0.1): https://semver.org/", checks.csv.Name))
+	}
+
+	// Check if its following the name convention
+	if len(strings.Split(checks.csv.Name, ".v")) < 2 {
```

Comment:
what if the length is greater than two? like `memcached-operator.v0.0.1.vfoo`, this warning wouldn't occur -- do we want to have the check where the `len != 2` instead -- is that safer?